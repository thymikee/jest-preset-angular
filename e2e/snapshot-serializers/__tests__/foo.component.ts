import { Component } from '@angular/core';
import { TestBed, waitForAsync } from '@angular/core/testing';

@Component({
  selector: 'foo',
  templateUrl: './foo.component.html',
  styleUrls: ['./foo.component.scss'],
  // we have to setup styles this way, since simple styles/styleUrs properties will be removed (jest does not unit test styles)
  styles: [
    `
      p {
        color: red;
      }
    `,
  ],
})
class FooComponent {}

test(
  'snapshot should work',
  waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [FooComponent],
    });

    const fixture = TestBed.createComponent(FooComponent);
    fixture.detectChanges();

    expect(fixture.debugElement.nativeElement).toMatchSnapshot();
  }),
);
