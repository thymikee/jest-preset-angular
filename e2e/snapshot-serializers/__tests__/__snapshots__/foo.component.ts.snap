// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should work 1`] = `
<div
  id="root0"
>
  <p>
     simple-with-styles works!

  </p>
</div>
`;
